apps/system/lockscreen/js/widgets/clock/lockscreen_clock_widget.js

```diff
@@ -50,6 +52,56 @@
     this.logger.debug('Clock updated', now);
   };
 
+  LockScreenClockWidget.prototype.updateAlarm =
+	function() {
+    this.fetchAlarmData()
+    .then((alarmData) => {
+      // if there is no alarm
+      if(!alarmData) {
+        this.resources.elements.alarm.classList.add('no-alarms');
+        this.resources.elements.alarm.classList.remove('has-alarm');
+        return;
+      }
+
+      var type = 'AM',
+          hour = parseInt(alarmData.hour, 10);
+
+      // Decide wheter to use AM or PM
+      if(hour > 12) {
+        type = 'PM';
+        hour = hour - 12;
+      }
+
+      var fullAlarmTime = hour + ':' + alarmData.minute + type;
+      this.resources.elements.alarmtime.textContent = fullAlarmTime;
+      this.resources.elements.alarm.classList.remove('no-alarms');
+      this.resources.elements.alarm.classList.add('has-alarm');
+      this.logger.debug('Alarm updated to :', fullAlarmTime);
+    });
+	};
+
+  /*
+   * Gets the Alarm Data from the DataStore.
+   */
+  LockScreenClockWidget.prototype.fetchAlarmData = function() {
+    return navigator.getDataStores('alarms')
+      .then((stores) => {
+        return [stores[0].getLength(), stores[0]];
+      }).then(([len, store]) => {
+        if(0 === len) {
+          return;
+        }
+        return store.get(1);
+      }).then((fetchedData = {}) => {
+        if(0 === Object.keys(fetchedData).length) {
+          return null;
+        }
+        return fetchedData;
+      }).catch(function(e){
+        console.error('Error occured while getting alarm data : ', e);
```

Comment:
I suggest doing that (the same logging and throwing). In fact, in a more complicated patch, I will design an `onPromiseError` with logging and re-throwing to reuse it in every `catch`. But since your patch only provides two chains that may have errors, to duplicate some code is OK.